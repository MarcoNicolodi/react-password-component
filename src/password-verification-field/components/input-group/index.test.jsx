import React from "react";
import renderer from "react-test-renderer";
import InputGroup from "./";

describe("InputGroup component", () => {
  const noop = () => {};
  it("should match snapshot when password is empty", () => {
    const tree = renderer.create(<InputGroup onChange={noop} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when password is weak", () => {
    const tree = renderer
      .create(<InputGroup onChange={noop} strength="weak" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when password is strong", () => {
    const tree = renderer
      .create(<InputGroup onChange={noop} strength="strong" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should match snapshot when password is very-strong", () => {
    const tree = renderer
      .create(<InputGroup onChange={noop} strength="very-strong" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should call onChange handler passed as props when input changes", () => {
    const onChangeMock = jest.fn();

    const tree = renderer.create(
      <InputGroup onChange={onChangeMock} strength="very-strong" />
    );

    const instance = tree.root;
    const input = instance.findByType("input");
    const fakeEvent = {
      target: {
        value: "my-password"
      }
    };
    input.props.onChange(fakeEvent);
    expect(onChangeMock).toHaveBeenCalledWith("my-password");
  });
});
