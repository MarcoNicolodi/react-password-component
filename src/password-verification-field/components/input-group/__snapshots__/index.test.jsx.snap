// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroup component should match snapshot when password is empty 1`] = `
<div
  className="input-control"
>
  <label
    htmlFor="password"
  >
    Password
    <input
      className={undefined}
      id="password"
      onChange={[Function]}
      placeholder="Digite sua senha"
      type="password"
    />
  </label>
</div>
`;

exports[`InputGroup component should match snapshot when password is strong 1`] = `
<div
  className="input-control"
>
  <label
    htmlFor="password"
  >
    Password
    <input
      className="strong"
      id="password"
      onChange={[Function]}
      placeholder="Digite sua senha"
      type="password"
    />
  </label>
</div>
`;

exports[`InputGroup component should match snapshot when password is very-strong 1`] = `
<div
  className="input-control"
>
  <label
    htmlFor="password"
  >
    Password
    <input
      className="very-strong"
      id="password"
      onChange={[Function]}
      placeholder="Digite sua senha"
      type="password"
    />
  </label>
</div>
`;

exports[`InputGroup component should match snapshot when password is weak 1`] = `
<div
  className="input-control"
>
  <label
    htmlFor="password"
  >
    Password
    <input
      className="weak"
      id="password"
      onChange={[Function]}
      placeholder="Digite sua senha"
      type="password"
    />
  </label>
</div>
`;
